App-Service-Environment-AzFirewall/azuredeploy.json

```diff
@@ -0,0 +1,326 @@
+                },
+                "parameters": {
+                    "location": {
+                        "value": "[parameters('location')]"
+                    },
+                    "vnetResourceName": {
+                        "value": "[parameters('vnetResourceName')]"
+                    },
+                    "vnetAddressRange": {
+                        "value": "[parameters('vnetAddressRange')]"
+                    },
+                    "aseSubnetName": {
+                        "value": "[parameters('aseSubnetName')]"
+                    },
+                    "aseSubnetAddressRange": {
+                        "value": "[parameters('aseSubnetAddressRange')]"
+                    },
+                    "aseNSGName": {
+                        "value": "[parameters('aseNSGName')]"
+                    },
+                    "aseRouteTableName": {
+                        "value": "[parameters('aseRouteTableName')]"
+                    },
+                    "azureFirewallName": {
+                        "value": "[parameters('azureFirewallName')]"
+                    },
+                    "azureFirewallSubnetAddressRange": {
+                        "value": "[parameters('azureFirewallSubnetAddressRange')]"
+                    },
+                    "azureFirewallRouteTableName": {
+                        "value": "[parameters('azureFirewallRouteTableName')]"
+                    },
+                    "azureFirewallPublicIP": {
+                        "value": "[parameters('azureFirewallPublicIP')]"
+                    },
+                    "deployAzureFirewall": {
+                        "value": "[parameters('deployAzureFirewall')]"
+                    },
+                    "tags": {
+                        "value": "[parameters('tags')]"
+                    }
+                }
+            },
+            "dependsOn": []
+        },
+        {
+            "apiVersion": "2015-08-01",
+            "type": "Microsoft.Web/hostingEnvironments",
+            "name": "[parameters('aseName')]",
+            "location": "[parameters('location')]",
+            "kind": "ASEV2",
+            "tags": {
+                "displayName": "Deploy ASE"
+            },
+            "properties": {
+                "name": "[parameters('aseName')]",
+                "location": "[parameters('location')]",
+                "ipSslAddressCount": 0,
+                "internalLoadBalancingMode": "[parameters('internalLoadBalancingMode')]",
+                "dnsSuffix": "[parameters('dnsSuffix')]",
+                "virtualNetwork": {
+                    "Id": "[variables('vnetID')]",
+                    "Subnet": "[parameters('aseSubnetName')]"
+                }
+            },
+            "dependsOn": [
+                "Microsoft.Resources/deployments/BuildOrUpdateASENetworking"
+            ]
+        },
+        {
+            "type": "Microsoft.web/serverfarms",
```

Comment:
Let's clean up the sort order of these properties, see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties - won't comment on the rest of them